### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by bkalika**

### CODE REVIEW

**Comment by bkalika** [comment]
[Homework #4. Tests. KALIKA BOHDAN](https://trello.com/c/VLMuJQcF/89-homework-4-tests-kalika-bohdan)

File: tests/tests.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+class TestRectanglePerimeter(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        print('Rectangle setUpClass')
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        print('Rectangle tearDownClass')
```
Comment (alchemixXx) [inline tests/tests.py]: useless methods

File: tests/tests.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        self.rect4 = Rectangle(-1, -1)
+        self.rect5 = Rectangle(0, 0)
+
+    def test_get_rectangle_perimeter(self):
+        self.assertEqual(self.rect1.get_rectangle_perimeter(), 8)
+        self.assertNotEqual(self.rect2.get_rectangle_perimeter(), 8)
+        self.assertLess(self.rect4.get_rectangle_perimeter(), 0)
+        self.assertFalse(self.rect5.get_rectangle_perimeter(), 0)
+        with self.assertRaises(TypeError):
+            self.rect3.get_rectangle_perimeter()
```
Comment (alchemixXx) [inline tests/tests.py]: it's better to separate it into two test: failed and passed

File: tests/tests.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        self.rect = Rectangle(2, 2)
+
+    def test_get_sum_of_corners(self):
+        self.assertEqual(self.rect.get_sum_of_corners(4), 360)
+        with self.assertRaises(ValueError):
+            self.rect.get_sum_of_corners(5)
+        with self.assertRaises(ValueError):
+            self.rect.get_sum_of_corners(-10)
+        with self.assertRaises(ValueError):
+            self.rect.get_sum_of_corners(0)
```
Comment (alchemixXx) [inline tests/tests.py]: the same as above. besides, you try to do it in a loop

File: tests/tests.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+
+class TestRectangleDiagonalAndCircumscribedCircle(unittest.TestCase):
+    def setUp(self) -> None:
+        self.rect1 = Rectangle(2, 2)
+        self.rect2 = Rectangle(0, 0)
+        self.rect3 = Rectangle(2, 3)
+        self.rect4 = Rectangle(-2, -2)
+        self.rect5 = Rectangle(10, 10)
+
```
Comment (alchemixXx) [inline tests/tests.py]: try to test one module in one test case

File: tests/tests.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        with self.assertRaises(ValueError):
+            self.rect2.get_rectangle_diagonal()
+
+    def test_get_radius_of_circumscribed_circle(self):
+        self.assertEqual(round(self.rect1.get_radius_of_circumscribed_circle(), 2), 1.41)
+        self.assertEqual(round(self.rect3.get_radius_of_circumscribed_circle(), 2), 1.8)
+        with self.assertRaises(ValueError):
+            self.rect2.get_radius_of_circumscribed_circle()
+        with self.assertRaises(ValueError):
+            self.rect4.get_radius_of_circumscribed_circle()
```
Comment (alchemixXx) [inline tests/tests.py]: the same as above

File: tests/tests.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+    def test_get_radius_of_inscribed_circle(self):
+        self.assertEqual(round(self.rect5.get_radius_of_inscribed_circle(), 2), float(10))
+        with self.assertRaises(ValueError):
+            self.rect2.get_radius_of_inscribed_circle()
+        with self.assertRaises(ValueError):
+            self.rect3.get_radius_of_inscribed_circle()
+        with self.assertRaises(ValueError):
+            self.rect4.get_radius_of_inscribed_circle()
```
Comment (alchemixXx) [inline tests/tests.py]: the same as above

File: tests/tests.py (hunk 1 above)
Comment (bkalika) [reply tests/tests.py]: deleted it

File: tests/tests.py (hunk 2 above)
Comment (bkalika) [reply tests/tests.py]: done

File: tests/tests.py (hunk 3 above)
Comment (bkalika) [reply tests/tests.py]: done

File: tests/tests.py (hunk 4 above)
Comment (bkalika) [reply tests/tests.py]: done

File: tests/tests.py (hunk 5 above)
Comment (bkalika) [reply tests/tests.py]: done

File: tests/tests.py (hunk 6 above)
Comment (bkalika) [reply tests/tests.py]: also done

File: tests/tests.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        self.assertFalse(self.rect1.get_rectangle_perimeter(), 0)
+        self.assertLessEqual(self.rect1.get_rectangle_perimeter(), 0)
+        self.assertLessEqual(self.rect1.get_rectangle_square(), 0)
+        self.assertLess(self.rect1.get_rectangle_square(), 2)
+        with self.assertRaises(ValueError):
+            self.rect1.get_rectangle_diagonal()
+        with self.assertRaises(ValueError):
+            self.rect1.get_radius_of_circumscribed_circle()
+        with self.assertRaises(ValueError):
+            self.rect1.get_radius_of_inscribed_circle()
```
Comment (alchemixXx) [inline tests/tests.py]: split it into different tests - failed and passed

File: tests/tests.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        self.rect2 = Rectangle(-2, -2)
+        self.assertLessEqual(self.rect2.get_rectangle_perimeter(), 0)
+        self.assertGreaterEqual(self.rect2.get_rectangle_square(), 0)
+        self.assertGreater(self.rect2.get_rectangle_square(), 2)
+        with self.assertRaises(ValueError):
+            self.rect2.get_rectangle_diagonal()
+        with self.assertRaises(ValueError):
+            self.rect2.get_radius_of_circumscribed_circle()
+        with self.assertRaises(ValueError):
+            self.rect2.get_radius_of_inscribed_circle()
```
Comment (alchemixXx) [inline tests/tests.py]: the same as above

File: tests/tests.py [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        self.rect1 = Rectangle(2, 2)
+        for i in [-10, 0, 5]:
+            with self.assertRaises(ValueError):
+                self.rect1.get_sum_of_corners(i)
+            with self.assertRaises(ValueError):
+                self.rect1.get_sum_of_corners(i)
+            with self.assertRaises(ValueError):
+                self.rect1.get_sum_of_corners(i)
+
+
```
Comment (alchemixXx) [inline tests/tests.py]: rename all tests: test_<instance>_<function>_<fail/pass>

**Pull request closed without merging**